src/compiler/checker.ts

```diff
@@ -10702,6 +10710,9 @@ namespace ts {
                     return keyofConstraintType;
                 }
                 if (t.flags & TypeFlags.IndexedAccess) {
+                    if (isGenericTupleType((<IndexedAccessType>t).objectType)) {
+                        return getIndexTypeOfType((<IndexedAccessType>t).objectType, IndexKind.Number);

```

Comment:
Likewise here, a stringlike index access (eg, "slice") can fetch array apparent members, which are not a component of the index.